### DISCUSSION TITLE
How to configure multiple rule.json in access-control.properties

### DISCUSSION THREAD

**Discussion by A-little-bit-of-data** [trinodb/trino#20128] [Q&A]
I read on the official website that there is this item in the access-control.properties configuration, security.config-file=etc/rules.json. Does anyone have such a need? There are many users. If it is written in a json file, the file will It is getting bigger and bigger and more and more difficult to control. I would like to ask everyone, can this rules.json be split into multiple parts? If so, how to configure it?
![image](https://github.com/trinodb/trino/assets/31812278/e523cd47-d159-43e8-9bb7-468b6e618e2f)

**Comment by gitvibav-abluva** [comment]
Hi @A-little-bit-of-data, I too have a need for this, but seems like this feature is not supported now. I tried contacting the community members in Slack and they told me about this. I think they should add these feature pretty soon, because it is much needed for users with thousands of policies. Therefore I am tagging @ebyhr and @mosabua.

**Comment by mosabua** [comment]
From what I know, nobody is working on that. In the end all the policies would still have to be collapsed into one set and then validated and such. As I mentioned on slack you can already do that now .. just have multiple files that you maintain in git or whereever and then have your deployment pipeline assembled on file from multiple input files.. 

If you want this actually as a feature in trino itself you will probably take up the work and send a pull request that implements that feature with tests and such.

**Reply by gitvibav-abluva** [reply]
With pleasure, I can implement this feature myself. But before that, if this [feature](https://github.com/trinodb/trino/issues/20152) is present, I am planning to implement that for my project.

**Reply by mosabua** [reply]
It should work .. as a result we updated the docs .. but you probably need to test it yourself. 

See https://trino.io/docs/current/security/built-in-system-access-control.html#multiple-access-control-systems and keep in mind that each access control is separate

**Reply by gitvibav-abluva** [reply]
I tried implementing it locally but it wasn't working for me.

I have `access-control.config-files=/etc/trino/file.properties,/etc/trino/file2.properties` in my config.properties.

Then file.properties and file2.properties with
```
access-control.name=file
security.config-file=/etc/trino/rules.json
security.refresh-period=1s

access-control.name=file
security.config-file=/etc/trino/rules2.json
security.refresh-period=1s
```
Then separate json file with policies. It wasn't showing any error though, the server got started but the policies weren't applied. Can you please elaborate on **keep in mind that each access control is separate**

**Reply by mosabua** [reply]
Please read https://trino.io/docs/current/security/built-in-system-access-control.html#multiple-access-control-systems

You might expect a different behavior than how it works..

**Reply by gitvibav-abluva** [reply]
Sure will do that
